cdisc_rules_engine/utilities/sdtm_utilities.py

```diff
@@ -374,3 +374,25 @@ def get_model_domain_metadata(model_details: dict, domain_name: str) -> dict:
 
 def replace_variable_wildcards(variables_metadata, domain):
     return [var["name"].replace("--", domain) for var in variables_metadata]
+
+
+def get_all_model_wildcard_variables(model_details: dict):
+    return {
+        classVariable["name"]
+        for cls in model_details.get("classes", [])
+        for classVariable in cls.get("classVariables", [])
+        if classVariable["name"].startswith("--")
```

Comment:
Should the wildcards come from the rule? Applies to the next function as well

Reply:
I don't think so. There are 2 different wildcards we are using:
- The wildcards used in the library model. This wildcard is always `--`, which is what we are using here and below. This means replace the wildcard with the current domain name.
- The wildcards specified by the rule and used by the `add_variables_wildcards` below. It will be the output wildcard in the dataset produced by the relrec matching. In the sample rule, I am using `__`. In the engine rule.py, I am defaulting this to `**` if it is not provided by the rule. This wildcard needs to be different from `--` because if we used the same wildcard, we would end up with something like `FATERM`, which is a variable in the left dataset, but not a variable in the right dataset.